ratelimits/limit.go

```diff
@@ -0,0 +1,142 @@
+package ratelimits
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/letsencrypt/boulder/config"
+	"github.com/letsencrypt/boulder/sa"
+	"github.com/letsencrypt/boulder/strictyaml"
+)
+
+type limit struct {
+	// Burst specifies maximum concurrent allowed requests at any given time. It
+	// must be greater than zero.
+	Burst int64
+
+	// Count is the number of requests allowed per period. It must be greater
+	// than zero.
+	Count int64
+
+	// Period is the duration of time in which the count (of requests) is
+	// allowed. It must be greater than zero.
+	Period config.Duration
+}
+
+func validateLimit(l limit) error {
+	if l.Burst <= 0 {
+		return fmt.Errorf("invalid burst '%d', must be > 0", l.Burst)
+	}
+	if l.Count <= 0 {
+		return fmt.Errorf("invalid count '%d', must be > 0", l.Count)
+	}
+	if l.Period.Duration <= 0 {
+		return fmt.Errorf("invalid period '%s', must be > 0", l.Period)
+	}
+	return nil
+}
+
+type limits map[string]limit
+
+// loadLimits marshals the YAML file at path into a map of limis.
+func loadLimits(path string) (limits, error) {
+	lm := make(limits)
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+	err = strictyaml.Unmarshal(data, &lm)
+	if err != nil {
+		return nil, err
+	}
+	return lm, nil
+}
+
+// parseOverrideNameId is broken out for ease of testing.
+func parseOverrideNameId(key string) (*Name, string, error) {
```

Comment:
Why does this return a *Name? Since Names are just int enums, a whole name is the same size as a pointer. No need to pass around pointers to them, we can copy them just fine, and then you don't have to be dereferencing all over the place in `loadAndParseOverrideLimits`.

Reply:
I actually went back and forth with myself on this one. Because the zero-value of the Name enumeration actually has a value I decided to make Name a pointer so that the zero value (nil) would be significant. I've gone ahead and added set the zero-value for Name to `Unknown`, following the example set by the status codes enum in the grpc-go package.